plugins/catalog-backend/openapi.yaml

```diff
@@ -0,0 +1,770 @@
+            - analysisSuggestedNoValue
+            - needsUserInput
+          description: The outcome of the analysis for this particular field
+        field:
+          type: string
+          description: |-
+            e.g. "spec.owner"? The frontend needs to know how to "inject" the field into the
+            entity again if the user wants to change it
+      required:
+        - description
+        - value
+        - state
+        - field
+      type: object
+    AnalyzeLocationGenerateEntity:
+      properties:
+        fields:
+          items:
+            $ref: '#/components/schemas/AnalyzeLocationEntityField'
+          type: array
+        entity:
+          $ref: '#/components/schemas/RecursivePartial_Entity_'
+      required:
+        - fields
+        - entity
+      type: object
+      description: |-
+        This is some form of representation of what the analyzer could deduce.
+        We should probably have a chat about how this can best be conveyed to
+        the frontend. It'll probably contain a (possibly incomplete) entity, plus
+        enough info for the frontend to know what form data to show to the user
+        for overriding/completing the info.
+    AnalyzeLocationResponse:
+      properties:
+        generateEntities:
+          items:
+            $ref: '#/components/schemas/AnalyzeLocationGenerateEntity'
+          type: array
+        existingEntityFiles:
+          items:
+            $ref: '#/components/schemas/AnalyzeLocationExistingEntity'
+          type: array
+      required:
+        - generateEntities
+        - existingEntityFiles
+      type: object
+    LocationInput:
+      properties:
+        type:
+          type: string
+        target:
+          type: string
+        presence:
+          type: string
+          enum:
+            - optional
+            - required
+      required:
+        - type
+        - target
+        - presence
+      type: object
+    SerializedError:
+      allOf:
+        - $ref: '#/components/schemas/JsonObject'
+        - properties:
+            code:
+              type: string
+              description: A custom code (not necessarily the same as an HTTP response code); may not be present
+            stack:
+              type: string
+              description: A stringified stack trace; may not be present
+            message:
+              type: string
+              description: The message of the exception that was thrown
+            name:
+              type: string
+              description: The name of the exception that was thrown
+          required:
+            - message
+            - name
+          type: object
+      description: The serialized form of an Error.
+  securitySchemes:
+    JWT:
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+
+paths:
+  /refresh:
+    post:
+      operationId: RefreshEntity
+      responses:
+        '200':
+          description: Refreshed
+      security:
+        - {}
```

Comment:
From https://stackoverflow.com/questions/47659324/how-to-specify-an-endpoints-authorization-is-optional-in-openapi-v3.